src/sentry/types/region.py

```diff
@@ -87,6 +86,21 @@ def to_url(self, path: str) -> str:
         """
         return urljoin(self.address, path)
 
+    def is_historic_monolith_region(self) -> bool:
+        """Check whether this is a historic monolith region.
+
+        In a monolith environment, there exists only the one monolith "region",
+        which is a dummy object.
+
+        In a siloed environment whose data was migrated from a monolith environment,
+        all region-scoped entities that existed before the migration belong to the
+        historic monolith region by default. Unlike in the monolith environment,
+        this region is not a dummy object, but nonetheless is subject to special
+        cases to ensure that legacy data is handled correctly.
```

Comment:
Please check this for factual accuracy.

Reply:
Seems reasonable.

Reply:
@GabeVillalobos Your slug work may want to use this method.

Reply:
Possibly, but the logic for appending a region prefix to slugs also needs to check if we're running in single-tenant mode so I'm not sure this would be comprehensive enough for that case?

Reply:
Good call.